cinema/serializers.py

```diff
@@ -1 +1,103 @@
-# write serializers here
+from rest_framework import serializers
+from rest_framework.fields import ReadOnlyField
+
+from cinema.models import Genre, CinemaHall, Actor, Movie, MovieSession
+
+
+class CinemaHallSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = CinemaHall
+        fields = (
+            "id",
+            "name",
+            "rows",
+            "seats_in_row",
+            "capacity",
+        )
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ("id", "name")
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Actor
+        fields = ("id", "first_name", "last_name", "full_name")
+
+
+class MovieSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Movie
+        fields = (
+            "id",
+            "title",
+            "description",
+            "duration",
+            "genres",
+            "actors",
+        )
+
+
+class MovieListSerializer(serializers.ModelSerializer):
+    genres = serializers.StringRelatedField(many=True)
+    actors = serializers.StringRelatedField(many=True)
+
+    class Meta:
+        model = Movie
+        fields = (
+            "id",
+            "title",
+            "description",
+            "duration",
+            "genres",
+            "actors",
+        )
+
+
+class MovieDetailSerializer(serializers.ModelSerializer):
+    genres = GenreSerializer(many=True, read_only=True)
+    actors = ActorSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = Movie
+        fields = (
+            "id",
+            "title",
+            "description",
+            "duration",
+            "genres",
+            "actors",
+        )
+
+
+class MovieSessionSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = MovieSession
+        fields = (
+            "show_time",
+            "movie",
+            "cinema_hall"
+        )
+
+
+class MovieSessionListSerializer(serializers.ModelSerializer):
+    movie = MovieListSerializer()
+    cinema_hall = CinemaHallSerializer()
+    movie_title = ReadOnlyField(source="movie.title")
+    cinema_hall_name = ReadOnlyField(source="cinema_hall.name")
+    cinema_hall_capacity = ReadOnlyField(source="cinema_hall.capacity")
+
+    class Meta:
+        model = MovieSession
+        fields = (
+            "id",
+            "show_time",
+            "movie_title",
+            "cinema_hall_name",
+            "cinema_hall_capacity",
+            "movie",
+            "cinema_hall",
```

Comment:
In the `MovieSessionListSerializer`, the fields `movie` and `cinema_hall` are included alongside their respective `ReadOnlyField` representations (`movie_title`, `cinema_hall_name`, `cinema_hall_capacity`). This duplication might be unnecessary if the intention is to only display the read-only fields. Consider removing `movie` and `cinema_hall` from the `fields` tuple if they are not needed.